### PR TITLE
Remove use of isTransit() function

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
**Description:**
The isTransit() function is unneeded because we can just use the `transitLeg` propery on legs. It was causing bugs with some of our custom modes.

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
I also deleted a file that wasn't imported anywhere. Did we mean to be using that?

**Comment by miles-grant-ibigroup** [comment]
> I also deleted a file that wasn't imported anywhere. Did we mean to be using that?

That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think

**Review by miles-grant-ibigroup — APPROVED** [review]
We're really on OTP2 now... eek!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Updating my review here to continue using the `isTransit` method in core-utils instead. Let's refactor that! This PR shouldn't be closed though I like the cleanup

**Comment by daniel-heppner-ibigroup** [comment]
> > I also deleted a file that wasn't imported anywhere. Did we mean to be using that?
> 
> That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think

Well it wasn't even exported in index.js

**Comment by miles-grant-ibigroup** [comment]
> > > I also deleted a file that wasn't imported anywhere. Did we mean to be using that?
> > 
> > 
> > That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think
> 
> Well it wasn't even exported in index.js

Let's just update core-utils instead

**Comment by daniel-heppner-ibigroup** [comment]
Will block this one until the OTP UI side is merged

EDIT: ready to go!

**Review by miles-grant-ibigroup — APPROVED** [review]
Swaggy

File: lib/actions/field-trip.js [hunk 1]
```diff
@@ -713,7 +712,7 @@ function checkValidityAndCapacity(state, request) {
       // check each individual trip to see if there aren't any trips in this
       // itinerary that are already in use by another field trip
       itinerary.legs
-        .filter((leg) => isTransit(leg.mode))
+        .filter((leg) => isTransitLeg(leg.mode))
```
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]:
```suggestion
        .filter((leg) => isTransitLeg(leg))
```

File: lib/components/narrative/default/default-itinerary.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ const ITINERARY_ATTRIBUTES = [
     order: 3,
     render: (itinerary, options) => {
       const leg = clone(itinerary.legs[0])
-      if (isTransit(leg.mode)) {
+      if (isTransitLeg(leg.mode)) {
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/default-itinerary.js]:
```suggestion
      if (isTransitLeg(leg)) {
```

File: lib/components/narrative/default/itinerary-description.tsx [hunk 3]
```diff
@@ -32,10 +32,7 @@ export function getMainItineraryModes({
   let transitMode
   itinerary.legs.forEach((leg, i) => {
     const { duration, mode, rentedBike, rentedVehicle } = leg
-    if (
-      (leg.transitLeg || isTransit(mode)) &&
-      duration > primaryTransitDuration
-    ) {
+    if (isTransitLeg(leg) && duration > primaryTransitDuration) {
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/itinerary-description.tsx:35]: I'm scared to remove the current check

File: lib/components/narrative/default/itinerary-summary.js [hunk 4]
```diff
@@ -93,16 +96,16 @@ export default class ItinerarySummary extends Component {
       if (
         i > 0 &&
         i < itinerary.legs.length - 1 &&
-        !leg.transitLeg &&
-        itinerary.legs[i - 1].transitLeg &&
-        itinerary.legs[i + 1].transitLeg
+        !isTransitLeg(leg) &&
+        isTransitLeg(itinerary.legs[i - 1]) &&
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/itinerary-summary.js:100]: I'd like to keep the old check maybe

File: lib/components/narrative/line-itin/LegIconWithA11y.tsx [hunk 5]
```diff
@@ -8,15 +10,17 @@ import { ComponentContext } from '../../../util/contexts'
...
-const LegIconWithA11y = (props: any) => {
+const LegIconWithA11y = (props: Props): JSX.Element => {
   // @ts-expect-error No type on ComponentContext
   const { LegIcon } = useContext(ComponentContext)
   const intl = useIntl()
   const { leg } = props
-  const { mode } = leg
-  const ariaLabel = isTransit(mode) ? getFormattedMode(mode, intl) : null
+  const { mode, transitLeg } = leg
+  const ariaLabel = transitLeg ? getFormattedMode(mode, intl) : null
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/LegIconWithA11y.tsx]: Why aren't we using the new method

File: lib/components/narrative/line-itin/realtime-time-column.tsx [hunk 6]
```diff
@@ -33,8 +33,7 @@ function RealtimeTimeColumn({ isDestination, leg }: Props): ReactElement {
   }
 
   const timeMillis = isDestination ? leg.endTime : leg.startTime
-  const isTransitLeg = isTransit(leg.mode)
-  const isRealtimeTransitLeg = isTransitLeg && leg.realTime
+  const isRealtimeTransitLeg = isTransitLeg(leg) && leg.realTime
 
   // For non-transit legs show only the scheduled time.
   if (!isTransitLeg) {
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/realtime-time-column.tsx]: This is now checking for the existence of a method

File: lib/components/narrative/narrative-itineraries.js [hunk 7]
```diff
@@ -423,9 +423,7 @@ class NarrativeItineraries extends Component {
         })
 
         // Identify whether an itinerary uses transit & sort into appropriate bucket
-        const transitLegs = cur.legs.filter(
-          (leg) => isTransit(leg.mode) | leg.transitLeg
-        )
+        const transitLegs = cur.legs.filter((leg) => isTransitLeg(leg.mode))
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/narrative-itineraries.js]:
```suggestion
        const transitLegs = cur.legs.filter((leg) => isTransitLeg(leg))
```

File: lib/util/itinerary.tsx [hunk 8]
```diff
@@ -81,7 +82,7 @@ export function getMinutesUntilItineraryStart(itinerary: Itinerary): number {
  * Gets the first transit leg of the given itinerary, or null if none found.
  */
 function getFirstTransitLeg(itinerary: Itinerary) {
-  return itinerary?.legs?.find((leg) => leg.transitLeg)
+  return itinerary?.legs?.find((leg) => isTransitLeg(leg))
```
Comment (miles-grant-ibigroup) [inline lib/util/itinerary.tsx]: Please do the same shorthand as above

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I trusted you! Post lunch I'm seeing this with new clairty

File: lib/components/narrative/default/itinerary-description.tsx (hunk 3 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:35]: time to get over your fears

File: lib/components/narrative/default/itinerary-summary.js (hunk 4 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-summary.js:100]: maybe, but maybe not

**Comment by daniel-heppner-ibigroup** [comment]
all the problems are fixed

File: lib/components/narrative/default/itinerary-description.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:35]: 🤬

File: lib/components/narrative/default/itinerary-summary.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/itinerary-summary.js:100]: 🤬

File: lib/actions/field-trip.js [hunk 9]
```diff
@@ -396,7 +396,7 @@ function makeSaveFieldTripItinerariesData(request, outbound, state) {
 
     const gtfsTripsForItinerary = []
     itinerary.legs
-      .filter((leg) => isTransit(leg.mode))
+      .filter((leg) => isTransitLeg(leg))
```
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]: We can use shorthand

File: lib/actions/field-trip.js [hunk 10]
```diff
@@ -713,7 +712,7 @@ function checkValidityAndCapacity(state, request) {
       // check each individual trip to see if there aren't any trips in this
       // itinerary that are already in use by another field trip
       itinerary.legs
-        .filter((leg) => isTransit(leg.mode))
+        .filter((leg) => isTransitLeg(leg))
```
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]: We can use the shorthand

File: lib/components/narrative/narrative-itineraries.js [hunk 11]
```diff
@@ -423,9 +423,7 @@ class NarrativeItineraries extends Component {
         })
 
         // Identify whether an itinerary uses transit & sort into appropriate bucket
-        const transitLegs = cur.legs.filter(
-          (leg) => isTransit(leg.mode) | leg.transitLeg
-        )
+        const transitLegs = cur.legs.filter((leg) => isTransitLeg(leg))
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/narrative-itineraries.js]: We can use the shorthand!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
It can be swaggier

**Review by miles-grant-ibigroup — APPROVED** [review]
Excited for the second reviewer

File: lib/components/narrative/default/itinerary-description.tsx [hunk 12]
```diff
@@ -68,7 +65,7 @@ export function ItineraryDescription({ itinerary }: Props): JSX.Element {
   let transitMode
   itinerary.legs.forEach((leg) => {
     const { duration, mode, rentedBike, rentedVehicle } = leg
-    if (isTransit(mode) && duration > primaryTransitDuration) {
+    if (isTransitLeg(leg) && duration > primaryTransitDuration) {
```
Comment (amy-corson-ibigroup) [inline lib/components/narrative/default/itinerary-description.tsx:68]: Maybe make this check a variable? It's used twice, here and 35

**Review by amy-corson-ibigroup — COMMENTED** [review]
Uh well it does work.... I will be honest I am missing the logic of why we need this in core utils? Why import a function when we can just use `leg.isTransit`

File: lib/components/narrative/default/itinerary-description.tsx (hunk 12 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:68]: I thought about that too but decided against it because it's a simple check anyway

**Review by amy-corson-ibigroup — APPROVED** [review]
I guess

**Pull request merged**